genetic_algorithm/qlearning.py

```diff
@@ -0,0 +1,254 @@
+"""
+Implementation of the Qlearning Algorithm.
+Wikipedia link: https://en.wikipedia.org/wiki/Q-learning
+Author: AryanRaj315
+"""
+
+import numpy as np
+import pickle
+
+with open("Qtable.txt", "rb") as f:
+    """
+    This function is responsible for loading Qtable.txt if already present
+    """
+    Q_table = pickle.load(f)
+
+
+def Action(state, Q_table):
+    """
+    This function returns the most suitable action value from the qtable.
+    """
+    action = np.argmax(Q_table[state])
+    if action is None:
+        action = randomAction()
+    return action
+
+
+def randomAction():
+    """
+    This function returns random action just in case the Qtable values return None.
+    """
+    action = np.random.choice([0, 1, 2, 3])
+    return action
```

Comment:
```suggestion
def next_best_action(state: int, Q_table: np.ndarray) -> int:
    """
    This function returns the most suitable action value from the qtable.
    
    >>> next_best_action(1, []) in [0, 1, 2, 3]
    True
    """
    action = np.argmax(Q_table[state])
    return action if action is not None else np.random.choice([0, 1, 2, 3])
```
Adds a proper function name in snake_case with Python type hints and a doctest.  Please read CONTRIBUTING.md.